java/maven.indexer/src/org/netbeans/modules/maven/indexer/ClassDependencyIndexCreator.java

```diff
@@ -168,50 +166,49 @@ private static boolean isArchiveFile(File jar) throws IOException {
     }
 
     static void search(String className, Indexer indexer, Collection<IndexingContext> contexts, List<? super ClassUsage> results) throws IOException {
-        String searchString = crc32base64(className.replace('.', '/'));
-        Query refClassQuery = indexer.constructQuery(ClassDependencyIndexCreator.FLD_NB_DEPENDENCY_CLASS.getOntology(), new StringSearchExpression(searchString));
-        TopScoreDocCollector collector = TopScoreDocCollector.create(NexusRepositoryIndexerImpl.MAX_RESULT_COUNT, null);
+        String searchString = crc32base32(className.replace('.', '/'));
```

Comment:
If I understand this correctly the indexed values are changed and thus an old index can't be reused. Can we make sure, that the old index is considered broken/obsolete/invalid and force a reindex or am I missing something?

Reply:
good question. I haven't tested that situation at all tbh since a NB upgrade does not copy cache files. Lucene 5 index would be incompatible with 8 anyway - Its likely the exiting code would reset the index on failure. 

I am going to copy the index from NB 13 to a test setup and see how it behaves.

Reply:
i ran the experiment as described above and it reindexed everything the next time the index was used.

Reply:
Thank you.